### PR TITLE
Fix admission webhook integration tests to filter out controller requests

### PR DESCRIPTION
**Pull request by jpbetz**
When implementing https://github.com/kubernetes/kubernetes/pull/77563 we noticed that requests from controllers were overwriting the requests originating from the test client in the request recorder, preventing the integration tests from validating the requests they intended to validate.

This PR filters out requests from everything except the test client by configuring the client with an easy to distinguish user name.

/kind cleanup

```release-note
NONE
```

cc @roycaihw @liggitt @caesarxuchao

### CODE REVIEW

**Comment by caesarxuchao** [comment]
@jpbetz which controllers are overwriting recorded requests? It might help me debug #76346

**Comment by liggitt** [comment]
flake #77573 
/retest

**Comment by jpbetz** [comment]
> @jpbetz which controllers are overwriting recorded requests? It might help me debug #76346

@roycaihw and I think the crd finalizer (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/finalizer/crd_finalizer.go#L159) was the one overwriting the delete requests of the CRDs. There were clearly other controllers submitting requests and I didn't look closely at.

**Comment by jpbetz** [comment]
> Joe Betz which controllers are overwriting recorded requests? It might help me debug #76346

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/finalizer/crd_finalizer.go#L159 , which @roycaihw helped me identify, was almost certainly the one overwriting the crd delete request recordings.

**Comment by jpbetz** [comment]
> @jpbetz  which controllers are overwriting recorded requests? It might help me debug #76346

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/finalizer/crd_finalizer.go#L159, which @roycaihw helped me identify, appears to be the one that was overwriting the delete requests in the recorder

**Comment by liggitt** [comment]
/lgtm cancel

needs rebase

**Comment by jpbetz** [comment]
Rebased

File: test/integration/apiserver/admissionwebhook/admission_test.go [hunk 1]
```diff
@@ -45,14 +45,16 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/wait"
 	dynamic "k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
```
Comment (liggitt) [inline test/integration/apiserver/admissionwebhook/admission_test.go]: already imported as clientset one line down

Comment (jpbetz) [reply test/integration/apiserver/admissionwebhook/admission_test.go]: thanks, fixed

**Pull request merged**